### ISSUE TITLE
Unclear description of C++ ABI

### ISSUE THREAD

**Issue by giordano** [pypackaging-native/pypackaging-native#16]
It isn't very clear what the C++ ABI discussed in the [Complex C++ dependencies](https://pypackaging-native.github.io/key-issues/native-dependencies/cpp_deps/) page is.  Since you mention "old" and "new" ABIs and refer to setting `_GLIBCXX_USE_CXX11_ABI` I presume you're talking about the [libstdc++ dual `std::string` ABI](https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_dual_abi.html), but that isn't intrinsic to the C++ language, rather to GCC's libstdc++ runtime, LLVM's libcxx library doesn't have this problem.

Another C++ ABI conflict may be the combination of different C++ runtime libraries (e.g. libstdc++ and libcxx), but that one doesn't seem to be explicitly addressed.

**Comment by h-vetinari** [comment]
There's a bit more discussion ABI (including the `std::string` case) in https://pypackaging-native.github.io/background/binary_interface/, but yeah, the main C++ deps page should do better in distinguishing the various sources of ABI differences.

We should also include "changing your C++ standard library" in the list of ways to break ABI on the background page.

**Comment by rgommers** [comment]
> I presume you're talking about the [libstdc++ dual `std::string` ABI](https://gcc.gnu.org/onlinedocs/libstdc++/manual/using_dual_abi.html), but that isn't intrinsic to the C++ language, rather to GCC's libstdc++ runtime, LLVM's libcxx library doesn't have this problem.

Yes indeed. Good points, thanks. The text should make it explicit that the issue stems from the runtime, and that package authors have no control over runtime selection and have to write their code in a way that it builds with GCC/Clang/MSVC at least, and more compilers for the most widely used packages.
